tests/js/web/DropZoneComponentTest.js

```diff
@@ -0,0 +1,928 @@
+            const clickSpy = sinon.spy();
+            const click = new Event('click');
+
+            $fileInput[0].addEventListener('click', clickSpy);
+            dropZoneComponent.processBrowseNode($browse[0], $fileInput[0]);
+            $browse[0].dispatchEvent(click);
+            expect(clickSpy).to.have.been.calledOnce;
+        });
+    });
+
+    describe('updateInfoState()', () => {
+        let browseStub;
+
+        beforeEach(() => {
+            browseStub = sinon.stub(dropZoneComponent, 'processBrowseNode');
+        });
+
+        afterEach(() => {
+            browseStub.reset();
+        });
+
+        it('should update the info node', () => {
+            dropZoneComponent.updateInfoState(0, 'foo');
+            expect($info.html()).to.equal('foo');
+        });
+
+        // partial stub
+        it('should re-attach the Browse Files handler', () => {
+            dropZoneComponent.updateInfoState(0, '<span class="browse"></span>');
+            expect(browseStub).to.have.been.calledOnce;
+        });
+    });
+
+    describe('updateDropZoneFiles()', () => {
+        let removeFileStub;
+
+        beforeEach(() => {
+            removeFileStub = sinon.stub(dropZoneComponent, 'removeFile');
+        });
+
+        afterEach(() => {
+            removeFileStub.restore();
+        });
+
+        it('should create a file wrapper if one does not exist', () => {
+            dropZoneComponent.updateDropZoneFiles(0);
+            expect($dropZone.find('.wrapper').length).to.equal(1);
+        });
+
+        it('should remove any validation messages', () => {
+            dropZoneComponent.updateDropZoneFiles(0);
+            expect(validation.clear).to.have.been.calledOnce;
+        });
+
+        it('should remove the wrapper if there are no files', () => {
+            $('<div class="wrapper"></div>').appendTo($dropZone);
+            instanceManager.getFiles.returns([]);
+            dropZoneComponent.updateDropZoneFiles(0);
+            expect($dropZone.find('.wrapper').length).to.equal(0);
+        });
+
+        it('should update the wrapper html with the files', () => {
+            dropZoneComponent.updateDropZoneFiles(0);
+            expect($dropZone.find('.wrapper').html()).to.equal('<div class="file"><a class="close" href="#"></a></div>');
+        });
+
+        // partial stub
+        it('should add the remove file handler to the files', () => {
+            const click = new Event('click');
+
+            dropZoneComponent.updateDropZoneFiles(0);
+            $html.find('.close')[0].dispatchEvent(click);
+            expect(removeFileStub).to.have.been.calledOnce;
+        });
+    });
+
+    describe('throwValidationError()', () => {
+        let partialStub;
+
+        beforeEach(() => {
+            partialStub = sinon.stub(dropZoneComponent, 'updateInfoState');
+        });
+
+        afterEach(() => {
+            partialStub.reset();
```

Comment:
No need for this reset, a new `DropZoneComponent` should be created above and the old one should be GC'd.